pringles/simulator/simulator.py

```diff
@@ -41,26 +40,33 @@ def __init__(self, process_result, main_log_path=None, output_path=None):
         self.process_result = process_result
         self.main_log_path = main_log_path
         self.output_path = output_path
-        if self.successful():
```

Comment:
This isn't needed because if the simulation fails an exception is thrown from the run_simulation method right?

Reply:
yeah. Maybe we should change that later, sometimes a simulation fails after simulating for a long time (because of a segfault or something), and the log and output files are generated anyway

Reply:
But if the simulation fails, the exit code of the process is not zero, and an exception is thrown. So the exception will bubble up, and no SimulationResult is generated. Am I right?

Reply:
Exactly